docs/cody/enterprise/model-configuration.mdx

```diff
@@ -0,0 +1,525 @@
+
+- Sourcegraph-provided models are enabled (`sourcegraph` is not set to `null`)
+- Requests to LLM providers are routed through the Cody Gateway (no `providerOverrides` field specified)
+- Sourcegraph-defined default models are used for Cody features (no `defaultModels` field specified)
+
+There are three main settings for configuring Sourcegraph-provided LLM models:
+
+|                         **Field**                         |                                            **Description**                                             |
+| --------------------------------------------------------- | ------------------------------------------------------------------------------------------------------ |
+| `endpoint`                                                | (Optional) The URL for connecting to Cody Gateway. The default is set to the production instance.          |
+| `accessToken`                                             | (Optional) The access token for connecting to Cody Gateway defaults to the current license key. |
+| [`modelFilters`](/cody/enterprise/model-configuration#model-filters) | (Optional) Filters specifying which models to include from Cody Gateway.                               |
+
+### Model Filters
+
+The `"modelFilters"` section allows you to control which Cody Gateway models are available to users of your Sourcegraph Enterprise instance. The following table describes each field:
+
+|   **Field**    |                                                                      **Description**                                                                       |
+| -------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `statusFilter` | Filters models based on their release status, such as stable, beta, deprecated or experimental. By default, all models available on Cody Gateway are accessible. |
+| `allow`        | An array of `modelRef`s specifying which models to include. Supports wildcards.                                                                            |
+| `deny`         | An array of `modelRef`s specifying which models to exclude. Supports wildcards.                                                                            |
+
+The following examples demonstrate how to use each of these settings together:
+
+```json
+"cody.enabled": true,
+"modelConfiguration": {
+  "sourcegraph": {
+    "modelFilters": {
+      // Only allow "beta" and "stable" models.
+      // Not "experimental" or "deprecated".
+      "statusFilter": ["beta", "stable"],
+
+      // Allow any models provided by Anthropic, OpenAI, Google and Fireworks.
+      "allow": [
+        "anthropic::*", // Anthropic models
+        "openai::*", // OpenAI models
+        "google::*", // Google Gemini models
+        "fireworks::*", // Autocomplete models like StarCoder and DeepSeek-V2-Coder hosted on Fireworks
```

Comment:
yeah if we're not recommending these models anymore I'm aligned

Reply:
@aramaraju, `fireworks::v1::deepseek-coder-v2-lite-base` is the default autocomplete model. Do we want to remove it from examples?

Reply:
@taras-yemets To be clear, we should have the examples. Here, I meant in the comment next to the `"fireworks::*` filter, instead of stating specific models, it should state something more generic (but shorter) "// Open-source models hosted by Sourcegraph on Fireworks.ai (typically used for autocomplete and Mistral Chat models)"

Reply:
Addressed in https://github.com/sourcegraph/docs/pull/735/commits/ec113707d2b58cadfe7c8765c0c397021b96288a